tests/backend/test_tf_ops.py

```diff
@@ -51,3 +51,25 @@ def test_reduced_density_matrix_fock_states(self, num_modes, setup_backend, cuto
         expected = tf.constant(expected)
 
         assert np.allclose(reduced_dm, expected, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize("modes", [[0], [1], [2], [0, 1], [0, 2], [1, 2], [0, 1, 2]])
+    def test_reduced_density_matrix_multiple_modes(self, setup_backend, cutoff, modes, tol):
+        """Test the reduced_density_matrix returns the correct reduced density matrices."""
+        state = np.zeros([cutoff, cutoff, cutoff])
+        state[0, 0, 0] = 1 / np.sqrt(2)
+        state[1, 1, 1] = 1 / np.sqrt(2)
+        state = np.multiply.outer(state, state.conj())
+
+        # transpose the state due to different conventions
+        state = state.transpose([0, 3, 1, 4, 2, 5])
+        state = tf.constant(state)
+
+        reduced_dm = reduced_density_matrix(state, modes, state_is_pure=False)
+
+        alpha_str = np.array(["ab", "cd", "ef"])
+        betha_str = np.array(["aa", "bb", "cc"])
+        betha_str[modes] = alpha_str[modes]
+        expected = np.einsum(f"{''.join(betha_str)}->{''.join(alpha_str[modes])}", state)
```

Comment:
Starting to get a bit hard to parse. Perhaps instead of trying to get too fancy, we can just parametrize in the decorator over the einsum strings instead?

Reply:
True, that would work as well. Would you prefer doing that rather than splitting up the einsum and adding a few extra lines and comments?

Reply:
yeah I think that might make it easier for future readers of this code :smiley: